compiler/sem/semexprs.nim

```diff
@@ -2687,17 +2744,32 @@ type
     paNone, paSingle, paTupleFields, paTuplePositions
 
 proc checkPar(c: PContext; n: PNode): TParKind =
-  if n.len == 0:
+  # xxx: rework the signature to retunr nkError and still figure out TParKind
+  case n.len:
+  of 0:
     result = paTuplePositions # ()
-  elif n.len == 1:
-    if n[0].kind == nkExprColonExpr: result = paTupleFields
-    elif n.kind == nkTupleConstr: result = paTuplePositions
-    else: result = paSingle         # (expr)
+  of 1:
+    result =
+      case n.kind
+      of nkTupleConstr:
+        case n[0].kind
+        of nkExprColonExpr:
+          paTupleFields
+        else:
+          paTuplePositions
+      else:
+        paSingle         # (expr)
```

Comment:
Very subtle change in logic, but doesn't matter since https://github.com/nim-lang/Nim/pull/13793

Reply:
That's so much better, decide once for the win. I'm hoping cleanup like this makes this stuff more obvious to spot. Adding that to my list of stuff to spruce up today.

Reply:
fixed this.

turns out there are still a number of places where nkPar with mroe than one element are passed in. I purposefully left the code ugly to make it easier to revisit.